src/groundlight/internalapi.py

```diff
@@ -122,19 +169,20 @@ def _add_label(self, image_query_id: str, label: str) -> dict:
 
         return response.json()
 
+    @_retry_http_connection
     def _get_detector_by_name(self, name: str) -> Detector:
         """Get a detector by name. For now, we use the list detectors API directly.
 
         TODO: Properly model this in the API, and generate SDK code for it.
         """
         url = f"{self.configuration.host}/v1/detectors?name={name}"
         headers = self._headers()
-        response = requests.request("GET", url, headers=headers)
 
-        if not is_ok(response.status_code):
-            raise InternalApiError(
-                f"Error getting detector by name '{name}' (status={response.status_code}): {response.text}",
-            )
+        try:
+            response = requests.request("GET", url, headers=headers)
+            response.raise_for_status()
```

Comment:
This is cleaner code, but is a functional change, and technically an incompatible change.  If somebody had code that caught `InternalApiError` it wouldn't work any more.